# <center>**Projet Prévision**: <font color=green>Développement des modéles et intérpretation des résultats (exercice1 et exercice2)

### <center>Elaborér par:<font color=blue> **Islem Ben Hassouna - 3BD2**

# **<font color=red>1)Exercice1**


### <font color=blue>Exploration des données:

In [156]:
library(readxl)
data <- read_excel("exercice 1.xlsx")

In [158]:
View(data)  

ID  sexe Region Age etat_civil N_educatif Membre_famille Revenu_famille
1   1  0    1      2   1          3           5             0             
2   2  0    1      4   2          3           5             0             
3   3  0    1      4   2          3          11             0             
4   5  0    1      3   2          3           4             0             
5   8  0    1      1   1          3           5             1             
6   9  0    1      4   3          3           4             1             
7  10  0    1      4   3          3           7             0             
8  11  0    1      4   2          3           8             0             
9  12  0    1      4   2          3           9             0             
10 15  0    1      4   2          3          10             0             
11 16  0    1      4   2          3           8             0             
12 17  0    1      4   2          3           7             0             
13 19  0    1      3   2          3           7             0             
14 20  0    1      3   2          3           6             0             
15 21  0    1      3   2          3           4             0             
16 22  0    1      1   1          3           3             0             
17 23  0    1      2   2          3           5             0             
18 24  0    1      1   1          2           6             0             
19 25  0    1      2   1          2           6             0             
20 26  0    1      1   1          2           6             0             
21 27  0    1      2   1          2           5             1             
22 28  0    1      4   1          1           4             1             
23 29  0    1      3   2          2           6             1             
24 31  0    1      2   1          2           7             1             
25 32  0    1      2   1          2           7             1             
26 34  0    1      4   2          2           9             0             
27 35  0    1      3   1          2          10             0             
28 36  0    1      2   1          2           4             0             
29 37  1    1      2   1          2           6             0             
30 38  0    1      4   2          2           6             0             
⋮  ⋮   ⋮    ⋮      ⋮   ⋮          ⋮          ⋮              ⋮             
62  90 0    2      1   1          3           5             0             
63  91 0    2      1   1          2           7             0             
64  92 0    2      1   1          3           4             1             
65  95 0    2      1   1          3          10             0             
66  96 0    2      2   1          3           5             0             
67  97 0    2      2   1          3          13             0             
68 105 0    2      4   2          2           6             0             
69 106 0    2      3   4          2           5             1             
70 107 0    2      4   3          2           4             1             
71 108 0    2      4   2          2           5             0             
72 109 0    2      4   2          2           8             0             
73 111 0    2      3   2          3           8             1             
74 113 0    2      2   2          2           5             0             
75 114 0    2      2   2          2          10             0             
76 115 0    2      2   1          3           6             1             
77 117 0    2      2   1          3           6             0             
78 122 0    2      2   1          3           5             1             
79 124 0    2      2   1          2           7             0             
80 135 0    2      2   1          2           6             0             
81 136 0    2      1   1          4           6             0             
82 137 0    2      2   1          2           7             0             
83 138 0    2      2   1    

In [43]:
summary(data) 

       ID              sexe             Region           Age       
 Min.   :  1.00   Min.   :0.00000   Min.   :1.000   Min.   :1.000  
 1st Qu.: 30.00   1st Qu.:0.00000   1st Qu.:1.000   1st Qu.:2.000  
 Median : 64.00   Median :0.00000   Median :1.000   Median :3.000  
 Mean   : 68.03   Mean   :0.06593   Mean   :1.352   Mean   :2.637  
 3rd Qu.:105.50   3rd Qu.:0.00000   3rd Qu.:2.000   3rd Qu.:4.000  
 Max.   :148.00   Max.   :1.00000   Max.   :2.000   Max.   :4.000  
                                                                   
   etat_civil      N_educatif    Membre_famille   Revenu_famille  
 Min.   :1.000   Min.   :1.000   Min.   : 3.000   Min.   :0.0000  
 1st Qu.:1.000   1st Qu.:2.000   1st Qu.: 5.000   1st Qu.:0.0000  
 Median :1.000   Median :2.000   Median : 6.000   Median :0.0000  
 Mean   :1.549   Mean   :2.538   Mean   : 6.418   Mean   :0.2747  
 3rd Qu.:2.000   3rd Qu.:3.000   3rd Qu.: 8.000   3rd Qu.:1.0000  
 Max.   :4.000   Max.   :5.000   Max.   :13.000   Max.

In [44]:
str(data) 

tibble [91 × 24] (S3: tbl_df/tbl/data.frame)
 $ ID                   : num [1:91] 1 2 3 5 8 9 10 11 12 15 ...
 $ sexe                 : num [1:91] 0 0 0 0 0 0 0 0 0 0 ...
 $ Region               : num [1:91] 1 1 1 1 1 1 1 1 1 1 ...
 $ Age                  : num [1:91] 2 4 4 3 1 4 4 4 4 4 ...
 $ etat_civil           : num [1:91] 1 2 2 2 1 3 3 2 2 2 ...
 $ N_educatif           : num [1:91] 3 3 3 3 3 3 3 3 3 3 ...
 $ Membre_famille       : num [1:91] 5 5 11 4 5 4 7 8 9 10 ...
 $ Revenu_famille       : num [1:91] 0 0 0 0 1 1 0 0 0 0 ...
 $ Revenu_Mensuelle     : num [1:91] 0 0 100 0 0 100 100 0 0 0 ...
 $ Redoublant           : num [1:91] 1 1 1 1 1 0 1 1 1 1 ...
 $ Prob_Enseignat       : num [1:91] 0 1 0 1 0 0 0 1 0 0 ...
 $ Aide_ecole           : num [1:91] 1 4 4 4 1 4 5 1 5 2 ...
 $ Rq_Comprtement       : num [1:91] 1 4 1 2 1 1 1 1 1 1 ...
 $ Relat_Parent         : num [1:91] 1 1 1 1 1 1 1 1 3 1 ...
 $ Niveau_educ_pere     : num [1:91] 2 2 2 1 3 2 2 1 2 2 ...
 $ Niveau_educ_mere     : nu

### <font color=green> 1) a) Expliquer la variable après rupture école en fonction des variables sexe, age, région, état civil, N-educatif, membre_famille, Revenu_famille, Revenu_mensuel, redoublant, prob_enseignant, Aide_école, Rq_comportement, Relation_parent, niveau educatif père, niveau éducatif mère, sign_école, endroit_fréquent, ph_famille, Ricule_prof et Ricule collègue.

### <font color=blue>**Regresion logistique multinomial:**

<font color=green> **Choix du modéle:** 
<font color=black> Dans ce travail, j'ai utilisé <font color=red> un modèle de régression logistique multinomiale <font color=black>afin de prédire<font color=red> la variable cible data$Apres_repture_ecole, qui est une variable catégorielle divisée en quatres classes distinctes : classe 1, classe 2, classe 3 et classe 4<font color=black>. La régression logistique multinomiale est particulièrement adaptée à ce type de problème, car elle permet de modéliser les relations entre plusieurs variables explicatives et une variable dépendante catégorielle à plus de deux niveaux.

Ainsi, le modèle a été ajusté pour estimer les probabilités d'appartenance de chaque observation aux différentes classes, permettant ainsi de classer les individus en fonction de leurs caractéristiques et de prédire dans quelle catégorie ils appartiendront, tout en prenant en compte les interactions entre les différentes classes.

In [268]:
library(nnet)

In [270]:
reg=multinom(data$Apres_repture_ecole~data$sexe+data$Age+data$Region+data$etat_civil+data$N_educatif+data$Membre_famille+data$Revenu_famille+data$Revenu_Mensuelle+data$Redoublant+data$Prob_Enseignat+data$Aide_ecole+data$Rq_Comprtement+data$Relat_Parent+data$Niveau_educ_pere+data$Niveau_educ_mere+data$Sign_ecole+data$Endroit_freq+data$Ph_famille+data$Ricule_Prof+data$Ricule_collegue)

# weights:  88 (63 variable)
initial  value 124.766493 
iter  10 value 78.622797
iter  20 value 63.239268
iter  30 value 55.258599
iter  40 value 49.846960
iter  50 value 47.101855
iter  60 value 43.719936
iter  70 value 43.481027
iter  80 value 43.465618
iter  90 value 43.465149
final  value 43.465144 
converged


In [272]:
summary(reg)

Warning message in sqrt(diag(vc)):
"Production de NaN"


Call:
multinom(formula = data$Apres_repture_ecole ~ data$sexe + data$Age + 
    data$Region + data$etat_civil + data$N_educatif + data$Membre_famille + 
    data$Revenu_famille + data$Revenu_Mensuelle + data$Redoublant + 
    data$Prob_Enseignat + data$Aide_ecole + data$Rq_Comprtement + 
    data$Relat_Parent + data$Niveau_educ_pere + data$Niveau_educ_mere + 
    data$Sign_ecole + data$Endroit_freq + data$Ph_famille + data$Ricule_Prof + 
    data$Ricule_collegue)

Coefficients:
  (Intercept) data$sexe    data$Age  data$Region data$etat_civil
2   -7.489772 245.87263  -0.4376716   -2.4084838        4.901879
3  -10.469941  47.08085   0.7066992    0.6411367        5.941687
4 -611.468711  82.99164 208.3453744 -225.7999524     -128.520080
  data$N_educatif data$Membre_famille data$Revenu_famille data$Revenu_Mensuelle
2   -5.771419e-01           0.2134464           0.4469855            0.00356250
3    8.525498e-03           0.1245541           0.7613484            0.00124591
4   -1.605677e+02

## <font color=blue> **Intérprétation:**

## <font color=green> logit(P(Y=k)/P(Y=1)​)=β0k​+β1k​⋅X1​+β2k​⋅X2​+⋯+βpk​⋅Xp​ ;  k∈{2,3,4}k∈{2,3,4}

## <font color=red> Catégorie 2 :
### logit⁡(P(Y=2)P(Y=1))=−7.489772+245.87263⋅(sexe)−0.4376716⋅(Age)−2.4084838⋅(Region)+4.901879⋅(etat_civil)−0.5771419⋅(N_educatif)+0.2134464⋅(Membre_famille)+0.4469855⋅(Revenu_famille)+0.0035625⋅(Revenu_Mensuelle)+0.6462154⋅(Redoublant)+1.3745384⋅(Prob_Enseignat)+0.5207737⋅(Aide_ecole)−1.2429099⋅(Rq_Comprtement)−0.6689233⋅(Relat_Parent)+0.4113848⋅(Niveau_educ_pere)+0.9943978⋅(Niveau_educ_mere)+0.2539833⋅(Sign_ecole)+0.1439979⋅(Endroit_freq)+0.8205266⋅(Ph_famille)−1.191701⋅(Ricule_Prof)+0.8176965⋅(Ricule_collegue)

## <font color=red> Catégorie 3 :
### log⁡it(P(Y=3)P(Y=1))=−10.469941+47.08085⋅(sexe)+0.7066992⋅(Age)+0.6411367⋅(Region)+5.941687⋅(etat_civil)+0.008525498⋅(N_educatif)+0.1245541⋅(Membre_famille)+0.7613484⋅(Revenu_famille)+0.00124591⋅(Revenu_Mensuelle)−0.2674273⋅(Redoublant)−0.3408284⋅(Prob_Enseignat)+0.5451506⋅(Aide_ecole)−0.8737298⋅(Rq_Comprtement)−0.716872⋅(Relat_Parent)−1.1034168⋅(Niveau_educ_pere)−0.09905559⋅(Niveau_educ_mere)+1.0602972⋅(Sign_ecole)−0.4305329⋅(Endroit_freq)−0.4167038⋅(Ph_famille)−3.501263⋅(Ricule_Prof)+2.1884588⋅(Ricule_collegue)


## <font color=red> Catégorie 4 :
### logit⁡(P(Y=4)P(Y=1))=−611.468711+82.99164⋅(sexe)+208.3453744⋅(Age)−225.7999524⋅(Region)−128.520080⋅(etat_civil)−160.5677⋅(N_educatif)+23.4909023⋅(Membre_famille)−32.9186956⋅(Revenu_famille)+0.42818081⋅(Revenu_Mensuelle)−187.7576202⋅(Redoublant)−38.2412999⋅(Prob_Enseignat)+62.2534163⋅(Aide_ecole)+110.9760625⋅(Rq_Comprtement)−90.3090799⋅(Relat_Parent)+175.6900377⋅(Niveau_educ_pere)+14.7043151⋅(Niveau_educ_mere)+50.5088759⋅(Sign_ecole)−9.8422942⋅(Endroit_freq)−36.6990838⋅(Ph_famille)+178.049957⋅(Ricule_Prof)−268.0826058⋅(Ricule_collegue)


Chaque coefficient associé à une variable dans le tableau indique l'effet de cette variable sur la probabilité d'appartenir à une catégorie donnée (par rapport à la catégorie de référence, ici catégorie 1).

   <font color=red> Un coefficient positif indique que lorsque la valeur de la variable augmente, la probabilité d'appartenir à cette catégorie augmente (par rapport à la catégorie 1).
   
<font color=red> Un coefficient négatif indique que lorsque la valeur de la variable augmente, la probabilité d'appartenir à cette catégorie diminue (par rapport à la catégorie 1).

<font color=purple>Par exemple :

    Pour la catégorie 2, le coefficient associé à sexe est 245.87. Cela signifie que le fait d'être de sexe masculin (ou la catégorie codée par 1 pour sexe) augmente considérablement la probabilité d'appartenir à la catégorie 2 par rapport à la catégorie 1.
    Pour la catégorie 4, le coefficient associé à Redoublant est -187.76, ce qui indique que les individus ayant redoublé ont une probabilité nettement plus faible d'appartenir à la catégorie 4 par rapport à la catégorie 1.



le modèle reg utilise une régression logistique multinomiale pour prédire la variable catégorielle Apres_repture_ecole, qui peut prendre quatre catégories différentes (1, 2, 3, ou 4), en fonction des variables explicatives fournies.

 <font color=purple> 1. Compréhension générale du modèle :

La régression logistique multinomiale permet de prédire la probabilité qu'un individu appartienne à une catégorie spécifique de la variable cible (Apres_repture_ecole), en fonction de plusieurs facteurs ou caractéristiques (comme sexe, âge, région, etc.).

Pour chaque catégorie (2, 3, 4), le modèle calcule un "logit", qui est le logarithme du rapport des probabilités entre la catégorie cible (2, 3, ou 4) et la catégorie de référence (ici la catégorie 1).



 Déviance résiduelle : 86.93 – Cela mesure dans quelle mesure le modèle s'adapte aux données. Plus ce chiffre est faible, meilleur est l'ajustement.
    AIC (Akaike Information Criterion) : 212.93 – Cela évalue la qualité globale du modèle en pénalisant les modèles trop complexes. Un AIC plus bas indique un modèle plus performant.

 <font color=purple> 2. Interprétation des catégories :

d'aprés les équations pour chaque catégorie (par rapport à la catégorie 1) :

<font color=purple> Catégorie 2 : <font color=black>
    Les individus de sexe masculin et ceux ayant des problèmes avec les enseignants ont une probabilité plus élevée d'appartenir à cette catégorie. Toutefois, un bon comportement (Rq_Comprtement) réduit cette probabilité (Le sexe, etat_civil et Prob_Enseignat semblent avoir un impact significatif).

<font color=purple> Catégorie 3 :<font color=black>
    Les jeunes adultes ou les individus dont les parents ont un faible niveau éducatif ont une probabilité accrue d'appartenir à cette catégorie. Par ailleurs, des relations positives avec les parents réduisent cette probabilité (sexe, etat_civil, et Rq_Comprtement jouent un rôle important).
    
<font color=purple> Catégorie 4 :<font color=black>
    Les coefficients extrêmes pour certaines variables comme Age, Region et Redoublant suggèrent que cette catégorie est fortement influencée par ces facteurs. Les personnes plus âgées, ayant redoublé plusieurs fois ou ayant des problèmes avec leurs enseignants, ont moins de chances d'appartenir à cette catégorie.

**=> Donc le modèle met en évidence les facteurs qui influencent significativement la probabilité qu'un individu tombe dans une catégorie particulière après avoir quitté l'école. Ces facteurs incluent le sexe, l'âge, le revenu familial, les redoublements, les relations parentales, et les problèmes rencontrés dans l'environnement scolaire.**

### <font color=green> b) Déterminer **la qualité du modèle.**

### <font color=blue>**Les testes d'hypothéses:**

In [101]:
install.packages("lmtest")

Installation du package dans 'C:/Users/USER/AppData/Local/R/win-library/4.3'
(car 'lib' n'est pas spécifié)



le package 'lmtest' a été décompressé et les sommes MD5 ont été vérifiées avec succés

Les packages binaires téléchargés sont dans
	C:\Users\USER\AppData\Local\Temp\RtmpMFabeY\downloaded_packages


In [53]:
library(lmtest)

#### <font color=brown>**test d'éxogéinété:** 

In [109]:
raintest(reg)


	Rainbow test

data:  reg
Rain = 0.81908, df1 = 45, df2 = 24, p-value = 0.7243


#### **Interprétation de test d'éxogéinété:** 
H0:indépendance entre x et e

H1:dépendance entre x et e

p-value=0.7243>0.05 => on accepte H0 : indépendance entre x et e

#### <font color=brown>**test d'autocorrélation des erreurs:** 

In [107]:
dwtest(reg)


	Durbin-Watson test

data:  reg
DW = 2.3257, p-value = 0.8594
alternative hypothesis: true autocorrelation is greater than 0


#### **Interprétation de test d'autocorrélation des erreurs:** 
H0:absence d'autocorrélation des erreurs

H1:autocorrélation des erreurs

p-value= 0.8594>0.05 => on accepte H0 : absence d'autocorrélation des erreurs

#### <font color=brown>**test normalité des erreurs:** 

In [115]:
shapiro.test(residuals(reg))


	Shapiro-Wilk normality test

data:  residuals(reg)
W = 0.81697, p-value < 2.2e-16


#### **Interprétation de test de normalité des erreurs:** 
H0:normalité des erreurs

H1:absence de normalité des erreurs

p-value= 2.2e-16<0.05 => on accepte H1 : absence de normalité des erreurs

### <font color=blue>**La matrice de corrélation:**

In [29]:
pred.prob=predict(reg,newdata=data)
print(pred.prob)

 [1] 1    1    2    2    2    3    4    2    2    2    3    2    2    2    3   
[16] 1    2    1    1    1    1    4    2    2    2    2    4    1    2    3   
[31] 3    3    3    <NA> 2    2    2    2    2    4    3    3    2    1    4   
[46] 3    2    2    4    3    4    1    3    1    2    1    4    1    3    1   
[61] 1    1    1    1    1    1    2    4    2    3    3    3    2    3    2   
[76] 1    1    1    1    1    1    1    1    1    4    4    1    1    1    1   
[91] 1   
Levels: 1 2 3 4


In [33]:
mc=table(data$Apres_repture_ecole,pred.prob)
print(mc)

   pred.prob
     1  2  3  4
  1 29  0  0  0
  2  3 23  5  0
  3  2  5 12  1
  4  0  0  0 10


In [75]:
# Calcul des métriques
accuracy <- sum(diag(mc)) / sum(mc)
precision <- diag(mc) / rowSums(mc)
recall <- diag(mc) / colSums(mc)
f1_score <- 2 * (precision * recall) / (precision + recall)

# Affichage des résultats
cat("Précision globale : ", accuracy, "\n")
cat("Précision par classe : ", precision, "\n")
cat("Rappel par classe : ", recall, "\n")
cat("F1-Score par classe : ", f1_score, "\n")

# Matrice de confusion normalisée
mc_normalized <- mc / sum(mc)
cat("Matrice de confusion normalisée : \n")
print(mc_normalized)


Précision globale :  0.8222222 
Précision par classe :  1 0.7419355 0.6 1 
Rappel par classe :  0.8529412 0.8214286 0.7058824 0.9090909 
F1-Score par classe :  0.9206349 0.779661 0.6486486 0.952381 
Matrice de confusion normalisée : 
   pred.prob
             1          2          3          4
  1 0.32222222 0.00000000 0.00000000 0.00000000
  2 0.03333333 0.25555556 0.05555556 0.00000000
  3 0.02222222 0.05555556 0.13333333 0.01111111
  4 0.00000000 0.00000000 0.00000000 0.11111111


### **Interprétation de la matrice de corrélation**
#### <font color=red>**Les résultats que vous avez obtenus montrent une bonne performance générale de votre modèle de régression logistique multinomiale(modéle de bonne qualité)**
1. Précision globale (0.822)
<font color=red> Cela signifie que 82,2 % des prédictions du modèle sont correctes. C'est un bon indicateur que le modèle fonctionne de manière efficace.

3. Précision par classe
    Classe 1 : Précision de 1.0 – Votre modèle fait des prédictions parfaites pour la classe 1.
    Classe 2 : Précision de 0.742 – Le modèle fait une erreur de classification dans environ 25.8 % des cas pour la classe 2.
    Classe 3 : Précision de 0.6 – Le modèle fait une erreur dans 40 % des cas pour la classe 3.
    Classe 4 : Précision de 1.0 – Le modèle prédit parfaitement les éléments de la classe 4.

<font color=red> Cela indique que votre modèle fait mieux pour certaines classes (notamment la classe 1 et la classe 4) que pour d'autres (comme la classe 3).

3. Rappel par classe

    Classe 1 : Rappel de 0.853 – Le modèle identifie 85,3 % des éléments de la classe 1.
    Classe 2 : Rappel de 0.821 – Le modèle identifie 82,1 % des éléments de la classe 2.
    Classe 3 : Rappel de 0.706 – Le modèle identifie 70,6 % des éléments de la classe 3.
    Classe 4 : Rappel de 0.909 – Le modèle identifie 90,9 % des éléments de la classe 4.

<font color=purple>Le modèle semble particulièrement performant pour identifier les éléments de la classe 4 et relativement bien pour la classe 1 et la classe 2, mais légèrement moins pour la classe 3.
4. F1-Score par classe

Le F1-score combine la précision et le rappel. Plus il est élevé, mieux c'est :

    Classe 1 : 0.921 – Excellente performance, équilibrée entre précision et rappel.
    Classe 2 : 0.780 – Performance moyenne, le modèle équilibre bien la précision et le rappel pour cette classe.
    Classe 3 : 0.649 – Le modèle a des difficultés à équilibrer précision et rappel pour cette classe.
    Classe 4 : 0.952 – Excellente performance, avec un bon équilibre entre précision et rappel.

<font color=red> 5. Matrice de confusion normalisée

La matrice de confusion normalisée montre la proportion des prédictions correctes et incorrectes par classe :

    Classe 1 : 32,2 % des prédictions sont correctement attribuées à la classe 1.
    Classe 2 : 25,6 % des observations réelles de la classe 2 sont correctement attribuées à cette classe, et une proportion non négligeable est incorrectement classée dans la classe 3.
    Classe 3 : 13,3 % des prédictions sont correctement attribuées à la classe 3, et des erreurs sont faites avec les classes 2 et 4.
    Classe 4 : 11,1 % des prédictions correctes pour la classe 4.


**=>Les points forts : Le modèle montre une bonne capacité à prédire correctement les classes 1 et 4, avec des scores de précision, de rappel et de F1 élevés pour ces classes.**

**=> Les points à améliorer : Le modèle semble moins performant pour la classe 3, avec une précision plus faible (60 %) et un rappel plus faible (70,6 %). Le modèle pourrait bénéficier de plus d'informations ou d'une meilleure modélisation pour cette classe.**
    

### <font color=green> 2) a) Expliquer la variable sensation après rupture école en fonction des variables sexe, age, région, état civil, N-educatif, membre_famille, Revenu_famille, Revenu_mensuel, redoublant, prob_enseignant, Aide_école, Rq_comportement, Relation_parent, niveau educatif père, niveau éducatif mère, sign_école, endroit_fréquent, ph_famille, Ricule_prof et Ricule collègue.

### <font color=blue>**Regresion logistique multinomial:**

<font color=green> Choix du modéle:<font color=black> Dans ce travail, j'ai utilisé <font color=red> un modèle de régression logistique multinomiale <font color=black> afin de prédire <font color=red>la variable cible data$Sensation_apres_ecole, <font color=black>qui est une variable catégorielle divisée en <font color=red> trois classes distinctes : classe 1, classe 2 et classe 3 <font color=black> La régression logistique multinomiale est particulièrement adaptée à ce type de problème, car elle permet de modéliser les relations entre plusieurs variables explicatives et une variable dépendante catégorielle à plus de deux niveaux.

Ainsi, le modèle a été ajusté pour estimer les probabilités d'appartenance de chaque observation aux différentes classes, permettant ainsi de classer les individus en fonction de leurs caractéristiques et de prédire dans quelle catégorie ils appartiendront, tout en prenant en compte les interactions entre les différentes classes.


In [160]:
reg1=multinom(data$Sensation_apres_ecole~data$sexe+data$Age+data$Region+data$etat_civil+data$N_educatif+data$Membre_famille+data$Revenu_famille+data$Revenu_Mensuelle+data$Redoublant+data$Prob_Enseignat+data$Aide_ecole+data$Rq_Comprtement+data$Relat_Parent+data$Niveau_educ_pere+data$Niveau_educ_mere+data$Sign_ecole+data$Endroit_freq+data$Ph_famille+data$Ricule_Prof+data$Ricule_collegue)

# weights:  66 (42 variable)
initial  value 98.875106 
iter  10 value 69.921757
iter  20 value 63.545892
iter  30 value 62.811185
iter  40 value 62.710519
iter  50 value 62.693622
iter  60 value 62.691463
final  value 62.691445 
converged


In [59]:
summary(reg1)

Call:
multinom(formula = data$Sensation_apres_ecole ~ data$sexe + data$Age + 
    data$Region + data$etat_civil + data$N_educatif + data$Membre_famille + 
    data$Revenu_famille + data$Revenu_Mensuelle + data$Redoublant + 
    data$Prob_Enseignat + data$Aide_ecole + data$Rq_Comprtement + 
    data$Relat_Parent + data$Niveau_educ_pere + data$Niveau_educ_mere + 
    data$Sign_ecole + data$Endroit_freq + data$Ph_famille + data$Ricule_Prof + 
    data$Ricule_collegue)

Coefficients:
  (Intercept)  data$sexe   data$Age data$Region data$etat_civil data$N_educatif
2  -2.0092479 -15.288989 1.70302770    2.247275      -2.1451832      -1.1150235
3  -0.9491642   1.817137 0.05406014   -1.968661       0.4986185      -0.6205035
  data$Membre_famille data$Revenu_famille data$Revenu_Mensuelle data$Redoublant
2         -0.06729289          -1.5735217          0.0023857860      -0.5856426
3         -0.14471002          -0.3600626          0.0007416636       3.8445603
  data$Prob_Enseignat data$Aide_eco

### <font color=purple> **Interprétation:**

La régression logistique multinomiale estime les probabilités d'appartenance à chaque classe k de la variable cible (ici k∈{1,2,3}k∈{1,2,3}) en fonction des variables prédictives. L'expression générale est la suivante :
#### <font color=purple>**log⁡(P(Y=k)P(Y=1))=β0k+∑j=1pβjkXj**

Expression du Modèle pour les Données

### <font color=red> Pour la classe Y=2 (comparée à Y=1) :
log⁡(P(Y=2)P(Y=1))=−2.009−15.289⋅sexe+1.703⋅Age+2.247⋅Region−2.145⋅etat_civil−1.115⋅N_educatif−0.067⋅Membre_famille−1.574⋅Revenu_famille+0.002⋅Revenu_Mensuelle−0.586⋅Redoublant−0.139⋅Prob_Enseignat−0.168⋅Aide_ecole+0.141⋅Rq_Comprtement−0.330⋅Relat_Parent−0.311⋅Niveau_educ_pere+0.944⋅Niveau_educ_mere−0.459⋅Sign_ecole+0.182⋅Endroit_freq+0.942⋅Ph_famille−0.139⋅Ricule_Prof−0.902⋅Ricule_collegue

### <font color=red>Pour la classe Y=3 (comparée à Y=1) :
log⁡(P(Y=3)P(Y=1))=−0.949+1.817⋅sexe+0.054⋅Age−1.969⋅Region+0.499⋅etat_civil−0.621⋅N_educatif−0.145⋅Membre_famille−0.360⋅Revenu_famille+0.001⋅Revenu_Mensuelle+3.845⋅Redoublant+1.683⋅Prob_Enseignat+0.308⋅Aide_ecole−0.021⋅Rq_Comprtement+0.095⋅Relat_Parent−1.924⋅Niveau_educ_pere+2.319⋅Niveau_educ_mere+0.272⋅Sign_ecole+0.095⋅Endroit_freq−0.608⋅Ph_famille−1.071⋅Ricule_Prof−0.869⋅Ricule_collegue


 Effets des variables : Chaque coefficient (β) mesure l'effet d'une unité d'augmentation de la variable correspondante sur le logarithme des rapports de probabilités (log⁡(P(Y=k)P(Y=1))log(P(Y=1)P(Y=k)​)).

<font color=purple> Un coefficient positif (β>0) indique que l'augmentation de la variable augmente la probabilité d'appartenir à la classe k par rapport à la classe de référence (Y=1).

<font color=purple>Un coefficient négatif (β<0) indique une diminution de cette probabilité.

Exemple avec sexe pour Y=2 : Le coefficient de sexe est fortement négatif (-15.289), ce qui signifie que l'appartenance au sexe codé comme 11 réduit considérablement la probabilité d'être dans la classe Y=2 par rapport à Y=1.

Variable Redoublant pour Y=3 : Le coefficient positif très élevé (3.845) indique que le fait d'être redoublant augmente fortement la probabilité d'appartenir à Y=3 par rapport à Y=1.

Effet du sexe :

    Pour la catégorie 2, le coefficient de sexe est extrêmement négatif (-15.29), ce qui indique que le sexe est fortement associé à une réduction de la probabilité d'appartenir à la catégorie 2. Cela pourrait suggérer que les individus d'un certain sexe (par exemple, masculin ou féminin, selon le codage des données) sont significativement moins susceptibles de ressentir cette sensation après l'école.
    Pour la catégorie 3, le coefficient est légèrement positif (1.82), indiquant une légère augmentation de la probabilité pour cette catégorie.

Autres variables :

    Âge : A un effet positif pour la catégorie 2 (1.70) et neutre pour la catégorie 3 (0.05). Cela pourrait indiquer que les étudiants plus âgés ressentent plus fréquemment une sensation spécifique après l'école.
    Redoublant : Ce facteur est très positif (3.84) pour la catégorie 3, ce qui suggère que les étudiants redoublants ont une probabilité élevée de ressentir une sensation correspondant à cette catégorie.
    Ricule des collègues : Le coefficient est négatif pour les deux catégories (-0.90 pour 2 et -0.87 pour 3), indiquant que les moqueries des collègues réduisent les probabilités d'appartenir aux catégories 2 et 3.

### <font color=green> b) Déterminer **la qualité du modèle**.

### <font color=blue>**Les testes d'hypothéses:**

#### <font color=brown>**test d'éxogéinété:** 

In [162]:
raintest(reg1)


	Rainbow test

data:  reg1
Rain = 0.95122, df1 = 45, df2 = 24, p-value = 0.5697


#### **Interprétation de test d'éxogéinété:** 
H0:indépendance entre x et e

H1:dépendance entre x et e

p-value=0.5697>0.05 => on accepte H0 : indépendance entre x et e

#### <font color=brown>**test d'autocorrélation des erreurs:** 

In [63]:
dwtest(reg1)


	Durbin-Watson test

data:  reg1
DW = 1.9168, p-value = 0.1877
alternative hypothesis: true autocorrelation is greater than 0


#### **Interprétation de test d'autocorrélation des erreurs:** 
H0:absence d'autocorrélation des erreurs

H1:autocorrélation des erreurs

p-value= 0.1877>0.05 => on accepte H0 : absence d'autocorrélation des erreurs

#### <font color=brown>**test normalité des erreurs:** 

In [65]:
shapiro.test(residuals(reg1))


	Shapiro-Wilk normality test

data:  residuals(reg1)
W = 0.96703, p-value = 7.302e-06


#### **Interprétation de test de normalité des erreurs:** 
H0:normalité des erreurs

H1:absence de normalité des erreurs

p-value=  7.302e-06<0.05 => on accepte H1 : absence de normalité des erreurs

### <font color=blue>**La matrice de confusion:**

In [141]:
pred.prob=predict(reg1,newdata=data)
print(pred.prob)

 [1] 1    1    1    3    1    1    1    3    1    2    3    2    3    3    1   
[16] 3    1    3    3    1    3    3    3    3    3    2    2    3    3    2   
[31] 3    3    2    <NA> 3    3    3    3    2    2    1    1    3    2    2   
[46] 2    1    2    2    3    1    2    3    3    2    1    2    3    3    1   
[61] 1    2    3    1    1    1    2    2    3    2    2    2    2    2    2   
[76] 2    2    1    2    2    1    2    3    2    2    2    1    1    1    1   
[91] 1   
Levels: 1 2 3


In [143]:
mc=table(data$Sensation_apres_ecole,pred.prob)
print(mc)

   pred.prob
     1  2  3
  1 21  4  7
  2  3 23  6
  3  3  6 17


In [145]:
# Calcul des métriques
accuracy <- sum(diag(mc)) / sum(mc)
precision <- diag(mc) / rowSums(mc)
recall <- diag(mc) / colSums(mc)
f1_score <- 2 * (precision * recall) / (precision + recall)

# Affichage des résultats
cat("Précision globale : ", accuracy, "\n")
cat("Précision par classe : ", precision, "\n")
cat("Rappel par classe : ", recall, "\n")
cat("F1-Score par classe : ", f1_score, "\n")

# Matrice de confusion normalisée
mc_normalized <- mc / sum(mc)
cat("Matrice de confusion normalisée : \n")
print(mc_normalized)


Précision globale :  0.6777778 
Précision par classe :  0.65625 0.71875 0.6538462 
Rappel par classe :  0.7777778 0.6969697 0.5666667 
F1-Score par classe :  0.7118644 0.7076923 0.6071429 
Matrice de confusion normalisée : 
   pred.prob
             1          2          3
  1 0.23333333 0.04444444 0.07777778
  2 0.03333333 0.25555556 0.06666667
  3 0.03333333 0.06666667 0.18888889


### <font color=blue>**l'interprétation:**
La matrice de confusion indique la performance de classification du modèle en comparant les valeurs prédites aux valeurs réelles. 

    Classes prédictives :
        Classe 1 : Le modèle a correctement classé 21 observations, mais en a mal classé 4 dans la classe 2 et 7 dans la classe 3.
        Classe 2 : Le modèle a correctement classé 23 observations, mais a mal classé 3 dans la classe 1 et 6 dans la classe 3.
        Classe 3 : Le modèle a correctement classé 17 observations, mais en a mal classé 3 dans la classe 1 et 6 dans la classe 2.

La matrice montre que le modèle a des difficultés à bien prédire la classe 3, car elle est souvent confondue avec les autres classes.
 Évaluation des métriques

    Précision globale : 67.8 %
   <font color=red> Cela signifie que le modèle prédit correctement la classe de Sensation_apres_ecole dans environ deux tiers des cas.(modéle acceptable)

    Précision par classe :
        Classe 1 : 65.63 %
        Classe 2 : 71.88 %
        Classe 3 : 65.38 %
        La précision est assez homogène entre les classes, bien que la classe 2 ait une performance légèrement meilleure. Cela indique que lorsque le modèle prédit une classe, la probabilité qu'il ait raison est relativement similaire entre les classes.

    Rappel par classe :
        Classe 1 : 77.78 %
        Classe 2 : 69.70 %
        Classe 3 : 56.67 %
        Le modèle identifie correctement les observations de la classe 1 dans 77.78 % des cas, mais il a plus de mal avec la classe 3 (56.67 %). Cela suggère que les caractéristiques associées à la classe 3 sont plus difficiles à distinguer.

    F1-Score par classe :
        Classe 1 : 71.19 %
        Classe 2 : 70.77 %
        Classe 3 : 60.71 %
        L'équilibre entre la précision et le rappel est le plus faible pour la classe 3, ce qui confirme que le modèle a des performances relativement moins bonnes pour cette catégorie.

    Matrice de confusion normalisée

La matrice normalisée représente la proportion des observations dans chaque catégorie :

    23.3 % des données totales sont bien classées dans la classe 1.
    25.6 % des données totales sont bien classées dans la classe 2.
    18.9 % des données totales sont bien classées dans la classe 3.

Cela montre que le modèle a une meilleure capacité à classer les observations dans les classes 1 et 2 que dans la classe 3.

**=>  Le modèle multinomial capture certaines relations importantes entre les variables indépendantes et la variable cible Sensation_apres_ecole. Toutefois, la performance globale (67.8 %) montre qu'il reste des marges d'amélioration, en particulier pour la classe 3, qui est la moins bien prédite.**

**=> Les variables comme le sexe, l'âge, le redoublement, et les relations sociales (comme les moqueries) semblent avoir un effet important sur la probabilité de ressentir une certaine sensation après l'école. Néanmoins, il serait judicieux de procéder à une analyse approfondie de la significativité statistique des coefficients et éventuellement d'explorer d'autres modèles ou approches pour améliorer la performance globale.**

### <font color=green> 3) a) Expliquer la variable violence en fonction des variables sexe, age, région, état civil, N-educatif, membre_famille, Revenu_famille, Revenu_mensuel, redoublant, prob_enseignant, Aide_école, Rq_comportement, Relation_parent, niveau educatif père, niveau éducatif mère, sign_école, endroit_fréquent, ph_famille, Ricule_prof et Ricule collègue.

### <font color=blue>**Regresion logistique binaire:**

<font color=green> Choix du modéle : <font color=black> Dans ce travail, j'ai utilisé <font color=red> un modèle de régression logistique binaire <font color=black> afin de prédire <font color=red>la variable cible data$Violence,<font color=black> qui est une variable catégorielle à deux niveaux distincts : <font color=red> présence ou absence de violence (0 ou 1). <font color=black>La régression logistique binaire est particulièrement adaptée à ce type de problème, car elle permet de modéliser les relations entre plusieurs variables explicatives et une variable dépendante binaire.

In [184]:
reg2=glm(data$Violence~data$sexe+data$Age+data$Region+data$etat_civil+data$N_educatif+data$Membre_famille+data$Revenu_famille+data$Revenu_Mensuelle+data$Redoublant+data$Prob_Enseignat+data$Aide_ecole+data$Rq_Comprtement+data$Relat_Parent+data$Niveau_educ_pere+data$Niveau_educ_mere+data$Sign_ecole+data$Endroit_freq+data$Ph_famille+data$Ricule_Prof+data$Ricule_collegue,family=binomial(logit))

In [71]:
summary(reg2)


Call:
glm(formula = data$Violence ~ data$sexe + data$Age + data$Region + 
    data$etat_civil + data$N_educatif + data$Membre_famille + 
    data$Revenu_famille + data$Revenu_Mensuelle + data$Redoublant + 
    data$Prob_Enseignat + data$Aide_ecole + data$Rq_Comprtement + 
    data$Relat_Parent + data$Niveau_educ_pere + data$Niveau_educ_mere + 
    data$Sign_ecole + data$Endroit_freq + data$Ph_famille + data$Ricule_Prof + 
    data$Ricule_collegue, family = binomial(logit))

Coefficients:
                       Estimate Std. Error z value Pr(>|z|)   
(Intercept)            6.068331   3.713106   1.634  0.10220   
data$sexe              1.524793   2.123152   0.718  0.47265   
data$Age              -1.055787   0.533032  -1.981  0.04762 * 
data$Region            1.002743   1.025685   0.978  0.32826   
data$etat_civil        0.023552   0.854491   0.028  0.97801   
data$N_educatif       -0.673330   0.624907  -1.077  0.28126   
data$Membre_famille   -0.061506   0.193548  -0.318  0.75065   
da

### <font color=blue>Formule du modéle:
### <font color=red> logit(P(Violence=1))=6.068+1.525⋅Sexe−1.056⋅Age+1.003⋅Region+0.024⋅Etat_civil−0.673⋅N_educatif−0.062⋅Membre_famille+1.283⋅Revenu_famille−0.005⋅Revenu_Mensuelle−1.211⋅Redoublant−2.011⋅Prob_Enseignat−0.197⋅Aide_ecole+0.509⋅Rq_Comprtement+0.418⋅Relat_Parent−0.504⋅Niveau_educ_pere−0.204⋅Niveau_educ_mere−0.094⋅Sign_ecole+0.392⋅Endroit_freq−1.159⋅Ph_famille+2.830⋅Ricule_Prof+0.984⋅Ricule_collegue

La probabilité de violence (P(Violence=1)P(Violence=1)) est donnée par la transformation logistique :
<font color=red> P(Violence=1)=1/(1+e−logit)

### <font color=blue>Intérprétation:
Variables significatives : Les variables ayant une influence significative sur la violence (p < 0.05) sont les suivantes :

    Âge : Le coefficient de -1.056 indique que l'âge a une relation négative avec la probabilité de violence. Plus l'individu est âgé, moins il est susceptible de montrer de la violence.
    Revenu mensuel : Un coefficient de -0.00496 signifie qu'une augmentation du revenu mensuel réduit légèrement la probabilité de violence.
    Problèmes avec les enseignants (Prob_Enseignat) : Un coefficient de -2.011 montre que les conflits ou problèmes avec les enseignants diminuent fortement la probabilité de violence.
    Phénomène familial (Ph_famille) : Un coefficient de -1.159 indique que les tensions ou conflits familiaux réduisent la probabilité de violence, de façon significative.
    Railleries des professeurs (Ricule_Prof) : Avec un coefficient de 2.829, cela montre que les railleries des professeurs augmentent significativement la probabilité de comportements violents.

Autres variables : Bien que les autres variables aient des coefficients, leurs p-valeurs montrent qu'elles ne sont pas significativement associées à la violence dans ce modèle.

Par exemple :
Si un individu a 20 ans, un sexe codé à 1, un revenu mensuel de 2000 unités, et a subi des railleries de professeurs (Ricule_Prof=1), vous insérez ces valeurs dans la formule pour obtenir P(Violence=1)P(Violence=1).

### <font color=green> b) Déterminer **la qualité du modèle.**

### <font color=blue>**Les testes d'hypothéses:**

#### <font color=brown>**test d'éxogéinété:** 

In [186]:
raintest(reg2)


	Rainbow test

data:  reg2
Rain = 1.0751, df1 = 45, df2 = 24, p-value = 0.4349


#### **Interprétation de test d'éxogéinété:** 
H0:indépendance entre x et e

H1:dépendance entre x et e

p-value=0.4349>0.05 => on accepte H0 : indépendance entre x et e

#### <font color=brown>**test d'autocorrélation des erreurs:** 

In [188]:
dwtest(reg2)


	Durbin-Watson test

data:  reg2
DW = 1.4902, p-value = 0.00152
alternative hypothesis: true autocorrelation is greater than 0


#### **Interprétation de test d'autocorrélation des erreurs:** 
H0:absence d'autocorrélation des erreurs

H1:autocorrélation des erreurs

p-value= 0.00152<0.05 => on accepte H1 : autocorrélation des erreurs

#### <font color=brown>**test normalité des erreurs:** 

In [77]:
shapiro.test(residuals(reg2))


	Shapiro-Wilk normality test

data:  residuals(reg2)
W = 0.93252, p-value = 0.0001621


#### **Interprétation de test de normalité des erreurs:** 
H0:normalité des erreurs

H1:absence de normalité des erreurs

p-value= 0.0001621<0.05 => on accepte H1 : absence de normalité des erreurs

#### <font color=brown>**test d'hétéroscédosité des erreurs:** 

In [79]:
bptest(reg2)


	studentized Breusch-Pagan test

data:  reg2
BP = 13.468, df = 20, p-value = 0.8564


#### **Interprétation de test d'hétéroscédastisité  des erreurs:** 
H0:homoscédastisité  des erreurs

H1:hétéroscédastisité  des erreurs

p-value= 0.8564>0.05 => on accepte H0 : homoscédastisité  des erreurs

### <font color=blue>**La matrice de confusion:**

In [158]:
pred.prob=predict(reg2,newdata=data)
print(pred.prob)

          1           2           3           4           5           6 
 0.88029446 -0.01494735 -5.85513268  0.19578559 -0.96194499 -0.84922268 
          7           8           9          10          11          12 
-1.00877197 -1.69772670 -4.20428950 -1.49392734 -1.26751766 -4.43011580 
         13          14          15          16          17          18 
-2.64455506 -1.79283259 -0.50506163  4.53775119 -0.10963352 -1.64192686 
         19          20          21          22          23          24 
-0.90475464  0.84375630 -0.57874206 -0.60992327  0.29857118 -1.71163777 
         25          26          27          28          29          30 
-2.10394284 -2.48802035 -3.82915219 -2.02141230  0.46290310 -5.45609969 
         31          32          33          34          35          36 
-4.08047669 -2.46593289 -3.34938505          NA  2.36780596  1.66116798 
         37          38          39          40          41          42 
-1.16323860 -0.43269011 -2.92238150 -5.36430366 -4.

In [161]:
pred.mod=factor(ifelse(pred.prob>0.5,"1","0"))

In [163]:
pred.mod

1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
   1    0    0    0    0    0    0    0    0    0    0    0    0    0    0    1 
  17   18   19   20   21   22   23   24   25   26   27   28   29   30   31   32 
   0    0    0    1    0    0    0    0    0    0    0    0    0    0    0    0 
  33   34   35   36   37   38   39   40   41   42   43   44   45   46   47   48 
   0 <NA>    1    1    0    0    0    0    0    0    0    0    0    0    0    0 
  49   50   51   52   53   54   55   56   57   58   59   60   61   62   63   64 
   1    0    0    0    0    0    0    0    1    1    0    1    0    0    0    1 
  65   66   67   68   69   70   71   72   73   74   75   76   77   78   79   80 
   1    1    0    0    0    0    0    0    0    0    0    1    0    0    0    1 
  81   82   83   84   85   86   87   88   89   90   91 
   1    1    0    0    0    0    1    1    1    0    0 
Levels: 0 1

In [165]:
mc=table(data$Violence,pred.mod)

In [173]:
print(mc)

   pred.mod
     0  1
  0 57  3
  1 14 16


In [175]:
# Calcul des métriques
accuracy <- sum(diag(mc)) / sum(mc)
precision <- diag(mc) / rowSums(mc)
recall <- diag(mc) / colSums(mc)
f1_score <- 2 * (precision * recall) / (precision + recall)

# Affichage des résultats
cat("Précision globale : ", accuracy, "\n")
cat("Précision par classe : ", precision, "\n")
cat("Rappel par classe : ", recall, "\n")
cat("F1-Score par classe : ", f1_score, "\n")

# Matrice de confusion normalisée
mc_normalized <- mc / sum(mc)
cat("Matrice de confusion normalisée : \n")
print(mc_normalized)


Précision globale :  0.8111111 
Précision par classe :  0.95 0.5333333 
Rappel par classe :  0.8028169 0.8421053 
F1-Score par classe :  0.870229 0.6530612 
Matrice de confusion normalisée : 
   pred.mod
             0          1
  0 0.63333333 0.03333333
  1 0.15555556 0.17777778


### <font color=purple>**Interprétation:**

D'aprés la matrice de confusion:

    Le modèle est très performant pour identifier les cas de non-violence, avec une spécificité de 95 %. Cela signifie qu'il commet très peu d'erreurs en classant les cas comme "non-violence".
    La précision de 84.21 % indique que lorsque le modèle prédit "violence", il a une forte probabilité d'avoir raison.

    Le rappel est relativement faible (53.33 %), ce qui signifie que le modèle manque une proportion importante des cas de violence (14 faux négatifs sur 30 cas réels de violence).
    Le F1-Score est également moyen (65.29 %), ce qui reflète un compromis entre précision et rappel.


<font color=red> Précision globale : Le modèle a une précision globale de 81.1%, ce qui est raisonnable pour ce type de données.(bonne qualité du modéle)
    Précision par classe :
    
 <font color=purple> Pour la classe "0" (non-violence), la précision est de 95%, ce qui montre que le modèle identifie bien les individus non violents.
 
<font color=purple> Pour la classe "1" (violence), la précision est de 53.3%, indiquant que le modèle a du mal à identifier précisément les cas de violence.

    Rappel par classe :
        Pour la classe "0", le rappel est de 80.3%, ce qui montre que le modèle identifie correctement la majorité des individus non violents.
        Pour la classe "1", le rappel est de 84.2%, signifiant que le modèle détecte une proportion importante des cas violents.
    F1-Score par classe :
        Pour la classe "0", le F1-score est de 87.0%, combinant précision et rappel de manière équilibrée.
        Pour la classe "1", le F1-score est de 65.3%, ce qui est plus faible, reflétant la difficulté à identifier correctement tous les cas de violence.

Matrice de confusion normalisée

La matrice de confusion normalisée montre que :

    63.3% des prédictions sont correctement classées comme "non-violence".
    17.8% des prédictions sont correctement classées comme "violence".
    Cependant, 15.5% des cas de violence sont mal classés comme "non-violence", ce qui peut poser un problème dans des applications sensibles.

**=> Ce modèle montre une bonne capacité à identifier les cas de non-violence, mais des limitations dans la détection des cas de violence. Les variables significatives comme l'âge, le revenu mensuel, les problèmes avec les enseignants, et les interactions avec les professeurs ou la famille donnent des indications intéressantes pour orienter des interventions ciblées. Cependant, des améliorations peuvent être apportées au modèle pour mieux équilibrer précision et rappel pour les cas de violence, par exemple en ajustant les seuils de décision ou en intégrant des variables explicatives supplémentaires.**

# **<font color=red>2)Exercice2**

### <font color=blue>**Exploration des données:**

In [91]:
library(readxl)
data1 <- read_excel("exercice 2.xlsx")

In [93]:
View(data1) 

Continente Dis Mag Value nombre total de décès Total Affecté
1   2          8                 12                     40      
2   3          7                476                 141277      
3   2          7                  3                     15      
4   4          7               1250                 134250      
5   3          7                 31                    200      
6   1          8                100                   2000      
7   2          7                400                  20350      
8   4          6                  2                  49350      
9   3          6                 38                  30253      
10  2          8                  4                   3006      
11  3          6                  1                  11050      
12  3          6                  1                  15123      
13  3          7                 11                    134      
14  2          7                 61                  40200      
15  3          6                  9                  16583      
16  1          7                177                  52272      
17  3          7                 19                   7618      
18  4          6                 24                    585      
19  2          7                 29                  88112      
20  2          8                 85                2348973      
21  2          7                 21                   3745      
22  1          8             242000                 164000      
23  2          8              23000                4993000      
24  1          6                573                 454755      
25  1          8                185                   3975      
26  2          7                 70                  40100      
27  3          7                 50                 600100      
28  2          8                 12                    200      
29  1          6                400                   2000      
30  2          8                160                    100      
⋮   ⋮          ⋮             ⋮                     ⋮            
79  3          7              22                     80400      
80  1          7             500                     38000      
81  3          6               1                      5005      
82  1          6              10                         5      
83  2          7               2                      5060      
84  1          7              12                     34503      
85  3          5               2                      1030      
86  2          7               4                      1524      
87  1          5              34                     18700      
88  2          5             250                     36200      
89  4          6             275                     21436      
90  1          6               1                        35      
91  2          7               6                     45238      
92  2          8             180                   1482275      
93  1          6              22                       300      
94  1          7              67                     16100      
95  3          5               3                      3060      
96  3          6              20                     45300      
97  2          7               5                      5112      
98  2          6               2                         6      
99  4          4               1                      3152      
100 1          6               2                       100      
101 1          8             939                   1270364      
102 2          8               9                      3850      
103 1          8               2                      5005      
104 1          6              34                     36048      
105 1          6              26                     43200      
106 1          7               5                      1500      
107 1          6              32                       619      
108 1          7             539       

In [15]:
summary(data1) 

   Continente    Dis Mag Value   nombre total de décès Total Affecté    
 Min.   :1.000   Min.   :1.000   Min.   :     1.0      Min.   :      5  
 1st Qu.:1.000   1st Qu.:6.000   1st Qu.:    11.0      1st Qu.:   2375  
 Median :2.000   Median :7.000   Median :    42.5      Median :  15188  
 Mean   :2.083   Mean   :6.676   Mean   :  3900.9      Mean   : 186036  
 3rd Qu.:3.000   3rd Qu.:7.000   3rd Qu.:   419.0      3rd Qu.:  79388  
 Max.   :5.000   Max.   :8.000   Max.   :242000.0      Max.   :4993000  
 Total des dommages ('000 US $)
 Min.   :     100              
 1st Qu.:    3112              
 Median :   20000              
 Mean   :  545449              
 3rd Qu.:  238550              
 Max.   :20000000              

In [16]:
str(data1) 

tibble [108 × 5] (S3: tbl_df/tbl/data.frame)
 $ Continente                    : num [1:108] 2 3 2 4 3 1 2 4 3 2 ...
 $ Dis Mag Value                 : num [1:108] 8 7 7 7 7 8 7 6 6 8 ...
 $ nombre total de décès         : num [1:108] 12 476 3 1250 31 100 400 2 38 4 ...
 $ Total Affecté                 : num [1:108] 40 141277 15 134250 200 ...
 $ Total des dommages ('000 US $): num [1:108] 500000 100000 7310 6000 3150 25000 125000 2000 8000 400 ...


### <font color=green>1) a) Expliquer la variable continente en fonction des variables Dis Mag value, nombre total de décès, total affecté et total des dommages.

### <font color=blue>**Regresion logistique multinomial:**

<font color=green> Choix du modéle: 
<font color=black> Dans ce travail, j'ai utilisé <font color=red> un modèle de régression logistique multinomiale <font color=black>afin de prédire<font color=red> la variable cible data1$Continente, qui est une variable catégorielle divisée en cinqs classes distinctes : classe 1, classe 2, classe 3 , classe 4  et classe 5<font color=black>. La régression logistique multinomiale est particulièrement adaptée à ce type de problème, car elle permet de modéliser les relations entre plusieurs variables explicatives et une variable dépendante catégorielle à plus de deux niveaux.

Ainsi, le modèle a été ajusté pour estimer les probabilités d'appartenance de chaque observation aux différentes classes, permettant ainsi de classer les individus en fonction de leurs caractéristiques et de prédire dans quelle catégorie ils appartiendront, tout en prenant en compte les interactions entre les différentes classes.

In [103]:
library(nnet)

In [105]:
colnames(data1) <- c("Continente", "Dis_Mag_Value", "Total_Deaths", "Total_Affected", "Total_Damages")
reg3 <- multinom(Continente ~ Dis_Mag_Value + Total_Deaths + Total_Affected + Total_Damages, data = data1)



# weights:  30 (20 variable)
initial  value 173.819295 
iter  10 value 165.148541
iter  20 value 131.900368
iter  30 value 125.614527
iter  40 value 125.098755
iter  50 value 124.569999
iter  60 value 123.846741
iter  70 value 122.842655
iter  80 value 122.716010
final  value 122.714668 
converged


In [27]:
summary(reg3)

Call:
multinom(formula = Continente ~ Dis_Mag_Value + Total_Deaths + 
    Total_Affected + Total_Damages, data = data1)

Coefficients:
  (Intercept) Dis_Mag_Value  Total_Deaths Total_Affected Total_Damages
2   -4.807929     0.6177011 -3.017864e-05   9.305018e-07  4.043856e-07
3    5.326481    -0.9018543 -3.933559e-04   1.298160e-06  2.230761e-07
4    7.680986    -1.4427079 -1.203217e-05   1.060519e-06  2.370130e-07
5   -6.456925     0.8655634 -2.439808e-02  -1.408477e-04 -1.795839e-04

Std. Errors:
   (Intercept) Dis_Mag_Value Total_Deaths Total_Affected Total_Damages
2 3.457665e-08  2.398805e-07 3.356182e-05   7.155007e-07  3.148590e-07
3 9.207998e-08  6.071898e-07 3.224619e-04   1.013857e-06  3.271382e-07
4 1.374329e-08  8.091810e-08 4.799140e-05   1.007641e-06  3.320683e-07
5 3.854395e-08  2.675433e-07 4.320014e-07   2.195122e-04  2.530911e-04

Residual Deviance: 245.4293 
AIC: 285.4293 


### <font color=purple>Formule du Modèle

Le modèle de régression multinomiale que vous avez ajusté peut être exprimé comme suit :
### <font color=purple>log(P(Y=k)P(Y=1))=β0,k+β1,k⋅Dis_Mag_Value+β2,k⋅Total_Deaths+β3,k⋅Total_Affected+β4,k⋅Total_Damages ; k={1,2,3,4,5}
Pour k=2 :
<font color=purple> log⁡(P(Y=2)P(Y=1))=−4.807929+0.6177011⋅Dis_Mag_Value−3.017864×10−5⋅Total_Deaths+9.305018×10−7⋅Total_Affected+4.043856×10−7⋅Total_Damages

Pour k=3 :
<font color=purple> log⁡(P(Y=3)P(Y=1))=5.326481−0.9018543⋅Dis_Mag_Value−3.933559×10−4⋅Total_Deaths+1.298160×10−6⋅Total_Affected+2.230761×10−7⋅Total_Damages

Pour k=4 :
<font color=purple> log⁡(P(Y=4)P(Y=1))=7.680986−1.4427079⋅Dis_Mag_Value−1.203217×10−5⋅Total_Deaths+1.060519×10−6⋅Total_Affected+2.370130×10−7⋅Total_Damages

Pour k=5 :
<font color=purple> log⁡(P(Y=5)P(Y=1))=−6.456925+0.8655634⋅Dis_Mag_Value−2.439808×10−2⋅Total_Deaths−1.408477×10−4⋅Total_Affected−1.795839×10−4⋅Total_Damages


### <font color=red>Interprétation:
Coefficients estimés :

Intercept : Les coefficients d'interception (par exemple, -4.807929 pour le continent 2) montrent la tendance moyenne de chaque continent en l'absence de toute autre variable explicative.

Dis_Mag_Value : Un coefficient positif (comme 0.6177011 pour le continent 2) signifie que plus la variable augmente, plus la probabilité d'appartenance au continent 2 augmente par rapport au continent de référence. À l'inverse, un coefficient négatif (comme -1.4427079 pour le continent 4) indique une diminution de la probabilité.

Total_Deaths : Un très petit coefficient négatif (comme -3.017864e-05 pour le continent 2) signifie que l'augmentation du nombre de décès réduit légèrement la probabilité d'être dans ce continent.

Total_Affected : Ce coefficient reflète l'influence du nombre total de personnes affectées. Les valeurs positives ou négatives indiquent l'augmentation ou la diminution de la probabilité d'appartenir à chaque continent.

Total_Damages : Ce coefficient montre l'effet des dommages économiques sur l'appartenance au continent. Un petit coefficient positif ou négatif a un impact moindre sur les probabilités.

### <font color=green> b) Déterminer **la qualité du modèle.**

### <font color=blue>**Les testes d'hypothéses:**

#### <font color=brown>**test d'éxogéniété:** 

In [87]:
raintest(reg3)


	Rainbow test

data:  reg3
Rain = 0.85191, df1 = 54, df2 = 49, p-value = 0.7179


#### **Interprétation de test d'éxogéinété:** 
H0:indépendance entre x et e

H1:dépendance entre x et e

p-value=0.7179>0.05 => on accepte H0 : indépendance entre x et e

#### <font color=brown>**test d'autocorrélation des erreurs:** 

In [89]:
dwtest(reg3)


	Durbin-Watson test

data:  reg3
DW = 1.7876, p-value = 0.1353
alternative hypothesis: true autocorrelation is greater than 0


#### **Interprétation de test d'autocorrélation des erreurs:** 
H0:absence d'autocorrélation des erreurs

H1:autocorrélation des erreurs

p-value= 0.1353>0.05 => on accepte H0 : absence d'autocorrélation des erreurs

#### <font color=brown>**test normalité des erreurs:** 

In [91]:
shapiro.test(residuals(reg3))


	Shapiro-Wilk normality test

data:  residuals(reg3)
W = 0.8712, p-value < 2.2e-16


#### **Interprétation de test de normalité des erreurs:** 
H0:normalité des erreurs

H1:absence de normalité des erreurs

p-value<2.2e-16<0.05 => on accepte H1 : absence de normalité des erreurs

### <font color=blue>**La matrice de confusion:**

In [29]:
pred.prob=predict(reg3,newdata=data1)
print(pred.prob)

  [1] 2 1 1 1 1 2 1 1 1 2 1 1 1 1 1 1 1 1 1 2 1 1 2 3 2 1 2 2 1 2 1 1 1 3 1 1 3
 [38] 2 1 1 4 1 1 3 1 1 1 1 1 2 1 1 1 3 1 1 1 1 1 1 1 2 2 2 2 2 1 1 1 3 3 2 1 2
 [75] 3 1 1 1 1 1 3 3 1 1 3 1 3 2 1 1 1 2 1 1 3 1 2 1 4 1 2 2 2 1 1 1 3 1
Levels: 1 2 3 4 5


In [31]:
mc=table(data1$Continente,pred.prob)
print(mc)

   pred.prob
     1  2  3  4  5
  1 29  7  5  0  0
  2 18 13  1  0  0
  3 13  3  6  0  0
  4  6  1  2  2  0
  5  2  0  0  0  0


In [33]:
# Calcul des métriques
accuracy <- sum(diag(mc)) / sum(mc)
precision <- diag(mc) / rowSums(mc)
recall <- diag(mc) / colSums(mc)
f1_score <- 2 * (precision * recall) / (precision + recall)

# Affichage des résultats
cat("Précision globale : ", accuracy, "\n")
cat("Précision par classe : ", precision, "\n")
cat("Rappel par classe : ", recall, "\n")
cat("F1-Score par classe : ", f1_score, "\n")

# Matrice de confusion normalisée
mc_normalized <- mc / sum(mc)
cat("Matrice de confusion normalisée : \n")
print(mc_normalized)


Précision globale :  0.462963 
Précision par classe :  0.7073171 0.40625 0.2727273 0.1818182 0 
Rappel par classe :  0.4264706 0.5416667 0.4285714 1 NaN 
F1-Score par classe :  0.5321101 0.4642857 0.3333333 0.3076923 NaN 
Matrice de confusion normalisée : 
   pred.prob
              1           2           3           4           5
  1 0.268518519 0.064814815 0.046296296 0.000000000 0.000000000
  2 0.166666667 0.120370370 0.009259259 0.000000000 0.000000000
  3 0.120370370 0.027777778 0.055555556 0.000000000 0.000000000
  4 0.055555556 0.009259259 0.018518519 0.018518519 0.000000000
  5 0.018518519 0.000000000 0.000000000 0.000000000 0.000000000


### <font color=blue>Interprétation:

<font color=red>Précision globale : 46.3% des prédictions faites par le modèle sont correctes.(qualité du modéle faible)

Précision par classe :

Continent 1 : 70.7% des événements classés comme appartenant au continent 1 sont corrects.
Continent 2 : 40.6% des événements classés comme appartenant au continent 2 sont corrects.
Continent 3 : 27.3% des événements classés comme appartenant au continent 3 sont corrects.
Continent 4 : 18.2% des événements classés comme appartenant au continent 4 sont corrects.
Continent 5 : 0%, ce qui signifie que le modèle n'a pas réussi à classer correctement cet ensemble.
Rappel par classe : Le rappel indique combien des événements réels dans chaque continent ont été correctement identifiés.

Le modèle a un rappel particulièrement faible pour les continents 3 et 5.
F1-Score : Cette métrique combine la précision et le rappel. Un score élevé indique un meilleur équilibre entre les deux. Pour les continents 1 et 2, les F1-scores sont raisonnables, mais pour les autres, ils sont très faibles, en particulier pour le continent 5 où aucune prédiction correcte n’a été faite. 

**=>Le modèle présente des résultats raisonnables pour certaines classes (continents), notamment le continent 1. Cependant, il a du mal à prédire correctement certains autres continents, particulièrement le continent 5, ce qui est reflété dans ses faibles scores de précision et de rappel.**

**=>le modèle prédit la probabilité d'appartenance à un continent donné en fonction des caractéristiques comme la magnitude des désastres, le nombre total de morts, les personnes affectées et les dommages. La précision globale est modérée, et certaines classes sont mieux prédictibles que d'autres. Les variables comme "Dis_Mag_Value" ont un effet important pour certaines classes, tandis que d'autres comme "Total_Deaths" ont un effet négligeable. Des ajustements ou des tests supplémentaires (par exemple, une validation croisée ou des ajustements des variables) pourraient améliorer les performances globales.**


### <font color=green>2) a) Expliquer la variable Dis Mag value en fonction des variables nombre total de décès, total affecté et total des dommages.

### <font color=blue>**Regresion logistique multinomial:**

<font color=green> Choix du modéle: 
<font color=black> Dans ce travail, j'ai utilisé <font color=red> un modèle de régression logistique multinomiale <font color=black>afin de prédire<font color=red> la variable cible data1$Continente, qui est une variable catégorielle divisée en six classes distinctes : classe 1, classe 4, classe 5 , classe 6 , classe 7  et classe 8<font color=black>. La régression logistique multinomiale est particulièrement adaptée à ce type de problème, car elle permet de modéliser les relations entre plusieurs variables explicatives et une variable dépendante catégorielle à plus de deux niveaux.

Ainsi, le modèle a été ajusté pour estimer les probabilités d'appartenance de chaque observation aux différentes classes, permettant ainsi de classer les individus en fonction de leurs caractéristiques et de prédire dans quelle catégorie ils appartiendront, tout en prenant en compte les interactions entre les différentes classes.

In [225]:
reg4 <- multinom(Dis_Mag_Value ~ Total_Deaths + Total_Affected + Total_Damages, data = data1)



# weights:  30 (20 variable)
initial  value 193.510023 
iter  10 value 178.341875
iter  20 value 133.918263
iter  30 value 127.265709
iter  40 value 126.230292
iter  50 value 125.705007
iter  60 value 125.098559
iter  70 value 124.351402
iter  80 value 124.027365
iter  90 value 123.578972
iter 100 value 123.309541
final  value 123.309541 
stopped after 100 iterations


In [227]:
summary(reg4)

Call:
multinom(formula = Dis_Mag_Value ~ Total_Deaths + Total_Affected + 
    Total_Damages, data = data1)

Coefficients:
  (Intercept)  Total_Deaths Total_Affected Total_Damages
4   1.4144650 -0.5933362445   6.511416e-06  0.0003920540
5   0.2818687 -0.0007285482   1.874325e-05  0.0003997089
6   1.3400011  0.0001790053   4.001649e-05  0.0003993648
7   1.7559870  0.0001754416   4.003299e-05  0.0003994797
8   0.4755223  0.0002000724   4.225506e-05  0.0003992066

Std. Errors:
   (Intercept) Total_Deaths Total_Affected Total_Damages
4 6.931299e-09 9.498022e-09   1.351225e-04  1.860743e-05
5 2.678236e-09 3.222084e-07   9.039424e-05  7.912646e-06
6 4.078630e-09 1.940440e-05   8.802337e-05  7.915000e-06
7 6.059144e-09 1.818987e-05   8.802112e-05  7.913400e-06
8 1.872733e-09 1.364613e-05   8.802322e-05  7.917249e-06

Residual Deviance: 246.6191 
AIC: 286.6191 


### <font color=purple>**Formule du Modèle:**

Le modèle multinomial que vous avez ajusté est de la forme :
Dis_Mag_Value=β0+β1⋅Total_Deaths+β2⋅Total_Affected+β3⋅Total_Damages

 Classe 4 (Dis_Mag_Value = 4) :
<font color=red> log⁡(P(Classe 4)P(Référence))=1.4144650−0.5933362445×Total_Deaths+6.511416×10−6×Total_Affected+0.0003920540×Total_Damages

Classe 5 (Dis_Mag_Value = 5) :
<font color=red> log⁡(P(Classe 5)P(Référence))=0.2818687−0.0007285482×Total_Deaths+1.874325×10−5×Total_Affected+0.0003997089×Total_Damages

Classe 6 (Dis_Mag_Value = 6) :
<font color=red> log⁡(P(Classe 6)P(Référence))=1.3400011+0.0001790053×Total_Deaths+4.001649×10−5×Total_Affected+0.0003993648×Total_Damages

Classe 7 (Dis_Mag_Value = 7) :
<font color=red> log⁡(P(Classe 7)P(Référence))=1.7559870+0.0001754416×Total_Deaths+4.003299×10−5×Total_Affected+0.0003994797×Total_Damages

Classe 8 (Dis_Mag_Value = 8) :
<font color=red> log⁡(P(Classe 8)P(Référence))=0.4755223+0.0002000724×Total_Deaths+4.225506×10−5×Total_Affected+0.0003992066×Total_Damages

### <font color=purple>Interprétation des coéfficients:

Classe 4 :

    Total_Deaths : Le coefficient négatif (-0.5933362445) indique qu'une augmentation du nombre de décès est associée à une probabilité réduite d'appartenir à la classe 
    Total_Affected : Le coefficient très petit (6.511416 × 10⁻⁶) montre une influence négligeable de l'impact total des affectés sur la classe 4.
    Total_Damages : Le coefficient positif (0.0003920540) indique que les dommages financiers plus importants augmentent la probabilité d'une observation dans la classe 
Classe 5 :

    Total_Deaths : Le coefficient très proche de zéro (-0.0007285482) montre qu'il n'y a quasiment aucune relation linéaire avec le nombre de décès.
    Total_Affected : Un léger coefficient positif (1.874325 × 10⁻⁵) suggère une légère influence positive des affectés sur la classe 5.
    Total_Damages : Le coefficient positif (0.0003997089) indique que des dommages plus élevés sont associés à une probabilité plus élevée d'appartenir à la classe 5.

Classe 6 :

    Total_Deaths : Un petit coefficient positif (0.0001790053) montre une très faible corrélation positive avec le nombre de décès.
    Total_Affected : Le coefficient positif (4.001649 × 10⁻⁵) indique une légère influence positive de l'impact des affectés sur la classe 6.
    Total_Damages : Un coefficient positif (0.0003993648) indique que les dommages financiers ont un impact positif sur la classe 6.

Classe 7 :

    Total_Deaths : Un petit coefficient positif (0.0001754416) suggère une très faible influence positive des décès sur la classe 7.
    Total_Affected : Le coefficient positif (4.003299 × 10⁻⁵) montre une légère corrélation positive avec le nombre de personnes affectées.
    Total_Damages : Le coefficient positif (0.0003994797) montre que des dommages plus élevés augmentent la probabilité de la classe 7.

Classe 8 :

    Total_Deaths : Un petit coefficient positif (0.0002000724) suggère une faible corrélation positive avec les décès.
    Total_Affected : Le coefficient positif (4.225506 × 10⁻⁵) montre une légère relation positive avec les personnes affectées.
    Total_Damages : Le coefficient positif (0.0003992066) indique une légère influence positive des dommages financiers.

### <font color=green> b) Déterminer **la qualité du modèle.**

### <font color=blue>**Les testes d'hypothéses:**

#### <font color=brown>**test d'éxogéniété:**

In [97]:
raintest(reg4)


	Rainbow test

data:  reg4
Rain = 0.55578, df1 = 54, df2 = 50, p-value = 0.9823


#### **Interprétation de test d'éxogéinété:** 
H0:indépendance entre x et e

H1:dépendance entre x et e

p-value=0.9823>0.05 => on accepte H0 : indépendance entre x et e

#### <font color=brown>**test d'autocorrélation des erreurs:** 

In [99]:
dwtest(reg4)


	Durbin-Watson test

data:  reg4
DW = 2.0193, p-value = 0.5433
alternative hypothesis: true autocorrelation is greater than 0


#### **Interprétation de test d'autocorrélation des erreurs:** 
H0:absence d'autocorrélation des erreurs

H1:autocorrélation des erreurs

p-value= 0.5433>0.05 => on accepte H0 : absence d'autocorrélation des erreurs

#### <font color=brown>**test de normalité des erreurs:** 

In [101]:
shapiro.test(residuals(reg4))


	Shapiro-Wilk normality test

data:  residuals(reg4)
W = 0.83341, p-value < 2.2e-16


#### **Interprétation de test de normalité des erreurs:** 
H0:normalité des erreurs

H1:absence de normalité des erreurs

p-value< 2.2e-16<0.05 => on accepte H1 : absence de normalité des erreurs

### <font color=blue>**La matrice de corrélation:**

In [41]:
pred.prob=predict(reg4,newdata=data1)
print(pred.prob)

  [1] 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 8 7 8 8 7 7 7 7 7 7 7 7 7 7 7 7 7 7
 [38] 7 7 7 7 7 7 7 7 7 7 7 7 8 7 7 7 7 7 7 7 8 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7
 [75] 7 7 7 7 7 7 7 7 7 7 7 7 7 5 7 7 7 8 7 7 7 7 7 7 7 7 8 7 7 7 7 7 7 7
Levels: 1 4 5 6 7 8


In [43]:
mc=table(data1$Dis_Mag_Value,pred.prob)
print(mc)

   pred.prob
     1  4  5  6  7  8
  1  0  0  0  0  1  0
  4  0  0  0  0  1  0
  5  0  0  1  0  7  0
  6  0  0  0  0 29  1
  7  0  0  0  0 48  0
  8  0  0  0  0 14  6


In [45]:
# Calcul des métriques
accuracy <- sum(diag(mc)) / sum(mc)
precision <- diag(mc) / rowSums(mc)
recall <- diag(mc) / colSums(mc)
f1_score <- 2 * (precision * recall) / (precision + recall)

# Affichage des résultats
cat("Précision globale : ", accuracy, "\n")
cat("Précision par classe : ", precision, "\n")
cat("Rappel par classe : ", recall, "\n")
cat("F1-Score par classe : ", f1_score, "\n")

# Matrice de confusion normalisée
mc_normalized <- mc / sum(mc)
cat("Matrice de confusion normalisée : \n")
print(mc_normalized)


Précision globale :  0.5092593 
Précision par classe :  0 0 0.125 0 1 0.3 
Rappel par classe :  NaN NaN 1 NaN 0.48 0.8571429 
F1-Score par classe :  NaN NaN 0.2222222 NaN 0.6486486 0.4444444 
Matrice de confusion normalisée : 
   pred.prob
              1           4           5           6           7           8
  1 0.000000000 0.000000000 0.000000000 0.000000000 0.009259259 0.000000000
  4 0.000000000 0.000000000 0.000000000 0.000000000 0.009259259 0.000000000
  5 0.000000000 0.000000000 0.009259259 0.000000000 0.064814815 0.000000000
  6 0.000000000 0.000000000 0.000000000 0.000000000 0.268518519 0.009259259
  7 0.000000000 0.000000000 0.000000000 0.000000000 0.444444444 0.000000000
  8 0.000000000 0.000000000 0.000000000 0.000000000 0.129629630 0.055555556


### <font color=purple>**Interprétation:**

Interprétation de la Matrice de Confusion (mc) :

La matrice de confusion (mc) vous permet d’évaluer la performance de votre modèle en termes de classification :

    Diagonale principale (vérités observées) : Montre combien de fois chaque classe a été correctement prédite.
        Par exemple, classe 6 a été correctement prédite 29 fois, et classe 8 a été correctement prédite 6 fois.

    Colonnes (prédictions) : Représente les classes prédits par le modèle.
        Par exemple, la classe 6 a été prédite 29 fois pour les vraies observations de classe 6, mais aussi 1 fois pour la classe 8.

    Lignes (réel) : Représente les classes réelles dans les données.
        Par exemple, pour la classe 7, le modèle a prédit cette classe 48 fois, mais seulement 1 fois pour la classe 8.

Interprétation des Métriques :

<font color=red> Précision Globale :
        La précision globale est de 50.93%, indiquant que 50.93% des prédictions étaient correctes.(qualité acceptable)

    Précision par Classe :
        Classe 5 : La précision pour la classe 5 est de 12.5%, ce qui signifie que seulement 12.5% des éléments prédits comme étant de la classe 5 étaient effectivement de la classe 5.
        Classe 7 : La précision pour la classe 7 est de 100%, ce qui signifie que toutes les prédictions pour la classe 7 étaient correctes.
        Classe 8 : La précision pour la classe 8 est de 30%, ce qui signifie que 30% des éléments prédits comme classe 8 étaient effectivement classés comme tels.

    Rappel par Classe :
        Classe 6 : Le rappel est de 100% pour la classe 6, ce qui signifie que toutes les instances de la classe 6 ont été correctement identifiées par le modèle.
        Classe 7 : Le rappel est de 48%, ce qui indique que seulement 48% des éléments réels de la classe 7 ont été capturés par le modèle.
        Classe 8 : Le rappel est de 85.7%, ce qui indique que 85.7% des éléments réels de la classe 8 ont été correctement identifiés.

    F1-Score par Classe :
        Classe 5 : L’F1-Score est de 22.22%, indiquant un compromis entre précision et rappel faible.
        Classe 7 : L’F1-Score est de 64.87%, ce qui est relativement bon pour la classe 7.
        Classe 8 : L’F1-Score est de 44.44%, indiquant une performance modérée.

Matrice de Confusion Normalisée :

La matrice normalisée montre la proportion d’observations dans chaque case par rapport au total de la matrice, ce qui vous permet de voir les erreurs relatives plutôt que les valeurs absolues. Par exemple, dans la case (5, 6), cela signifie que 0.2685 de la classe réelle 6 a été mal classée en classe 7.


**=> Total_Deaths (Total des Décès) : A un effet négatif dans les classes 4 et 5, suggérant que des catastrophes avec un plus grand nombre de décès sont moins susceptibles de correspondre à ces classes. A un effet positif mais faible dans les classes 6, 7, et 8.
=>Total_Affected (Total des Personnes Affectées) : A un effet positif sur la probabilité d’appartenir à toutes les classes, avec un impact particulièrement marqué dans les classes de magnitudes plus élevées.
=>Total_Damages (Total des Dommages) : A également un effet positif sur la probabilité d’appartenir à toutes les classes, avec un effet linéaire et similaire dans toutes les classes.
=> Ces interprétations montrent comment ces variables affectent la probabilité de classes spécifiques de catastrophes, et cela peut être utile pour comprendre les relations entre la taille d'une catastrophe (en termes de décès, affectés, et dommages) et la probabilité d'une certaine classification de magnitude.**



### <font color=green>3) a) Etudier la stationarité des variables:  Dis Mag value,  nombre total de décès,  total affecté et total des dommages.

### <font color=blue>**Les testes de stationnarité:**

In [49]:
install.packages("tseries")

Installation du package dans 'C:/Users/USER/AppData/Local/R/win-library/4.3'
(car 'lib' n'est pas spécifié)

installation des dépendances 'xts', 'TTR', 'quadprog', 'quantmod'




le package 'xts' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'TTR' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'quadprog' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'quantmod' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'tseries' a été décompressé et les sommes MD5 ont été vérifiées avec succés

Les packages binaires téléchargés sont dans
	C:\Users\USER\AppData\Local\Temp\RtmpMFabeY\downloaded_packages


In [97]:
library(tseries)


In [53]:
adf.test(data1$Dis_Mag_Value)

Warning message in adf.test(data1$Dis_Mag_Value):
"p-value smaller than printed p-value"



	Augmented Dickey-Fuller Test

data:  data1$Dis_Mag_Value
Dickey-Fuller = -4.2359, Lag order = 4, p-value = 0.01
alternative hypothesis: stationary


### <font color=blue>intérpretation:
#### H0:"Mag value n'est pas stationnaire"
#### H1:"Mag value est stationnaire"
#### p-value=0.01<0.05
#### Lag order = 4 : Cela indique que le modèle ADF utilise 4 retards (lags) dans son calcul. Cela peut être ajusté selon le nombre de périodes passées utilisées pour modéliser les relations temporelles dans les données.
#### <font color=red> On accepte H1 Donc la variabe Dis Mag value est stationnaire :la série temporelle est stationnaire d'ordre 0, c'est-à-dire qu'elle a une moyenne, une variance et une covariance constantes dans le temps.

In [109]:
adf.test(data1$Total_Deaths)

Warning message in adf.test(data1$Total_Deaths):
"p-value smaller than printed p-value"



	Augmented Dickey-Fuller Test

data:  data1$Total_Deaths
Dickey-Fuller = -4.8164, Lag order = 4, p-value = 0.01
alternative hypothesis: stationary


### <font color=blue>intérpretation:
#### H0:"Total_Deaths n'est pas stationnaire"
#### H1:"Total_Deaths est stationnaire"
#### p-value=0.01<0.05
#### Lag order = 4 : Cela indique que le modèle ADF utilise 4 retards (lags) dans son calcul. Cela peut être ajusté selon le nombre de périodes passées utilisées pour modéliser les relations temporelles dans les données.
#### <font color=red> On accepte H1 Donc la variabe Total_Deaths value est stationnaire :la série temporelle est stationnaire d'ordre 0, c'est-à-dire qu'elle a une moyenne, une variance et une covariance constantes dans le temps.

In [69]:
adf.test(data1$Total_Affected)


Warning message in adf.test(data1$Total_Affected):
"p-value smaller than printed p-value"



	Augmented Dickey-Fuller Test

data:  data1$Total_Affected
Dickey-Fuller = -4.0487, Lag order = 4, p-value = 0.01
alternative hypothesis: stationary


### <font color=blue>intérpretation:
#### H0:"Total_Affected n'est pas stationnaire"
#### H1:"Total_Affected est stationnaire"
#### p-value=0.01<0.05
#### Lag order = 4 : Cela indique que le modèle ADF utilise 4 retards (lags) dans son calcul. Cela peut être ajusté selon le nombre de périodes passées utilisées pour modéliser les relations temporelles dans les données.
#### <font color=red> On accepte H1 Donc la variabe Total_Affected value est stationnaire :la série temporelle est stationnaire d'ordre 0, c'est-à-dire qu'elle a une moyenne, une variance et une covariance constantes dans le temps.

In [72]:
adf.test(data1$Total_Damages)

Warning message in adf.test(data1$Total_Damages):
"p-value smaller than printed p-value"



	Augmented Dickey-Fuller Test

data:  data1$Total_Damages
Dickey-Fuller = -5.2938, Lag order = 4, p-value = 0.01
alternative hypothesis: stationary


### <font color=blue>intérpretation:
#### H0:"Total_Damages n'est pas stationnaire"
#### H1:"Total_Damages est stationnaire"
#### p-value=0.01<0.05
#### Lag order = 4 : Cela indique que le modèle ADF utilise 4 retards (lags) dans son calcul. Cela peut être ajusté selon le nombre de périodes passées utilisées pour modéliser les relations temporelles dans les données.
#### <font color=red> On accepte H1 Donc la variabe Total_Damages value est stationnaire :la série temporelle est stationnaire d'ordre 0, c'est-à-dire qu'elle a une moyenne, une variance et une covariance constantes dans le temps.

### <font color=green>b)  Expliquer la variable Dis Mag value en fonction des variables nombre total de décès, total affecté et total des dommages.


### <font color=blue>**Le modèle VAR (Vector AutoRegression)**

<font color=green> Choix du modéle: 
<font color=black> Dans ce travail, j'ai utilisé <font color=red> Le modèle VAR <font color=black>car tous les variables Dis Mag value, nombre total de décès, total affecté et total des dommages sont <font color=red> des variables stationnaire d'ordre 0 <font color=black>.Le modèle VAR (Vector AutoRegression) est utilisé pour modéliser et analyser les relations entre plusieurs séries temporelles interconnectées.

In [2]:
install.packages("vars")

Installation du package dans 'C:/Users/USER/AppData/Local/R/win-library/4.3'
(car 'lib' n'est pas spécifié)

installation des dépendances 'strucchange', 'urca', 'sandwich'




le package 'strucchange' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'urca' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'sandwich' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'vars' a été décompressé et les sommes MD5 ont été vérifiées avec succés

Les packages binaires téléchargés sont dans
	C:\Users\USER\AppData\Local\Temp\RtmpE7jJCw\downloaded_packages


In [6]:
library(vars)

### <font color=purple> Sélection de l’ordre du modèle VAR

Déterminez le nombre de retards (lags) à inclure dans le modèle à l’aide de critères comme AIC (Akaike Information Criterion) ou BIC (Bayesian Information Criterion).

In [16]:
# Sélection du lag optimal avec les données brutes
lag_selection <- VARselect(data1, lag.max = 10, type = "const")
print(lag_selection)


$selection
AIC(n)  HQ(n)  SC(n) FPE(n) 
     1      1      1      1 

$criteria
                  1            2            3            4            5
AIC(n) 7.596636e+01 7.610007e+01 7.623456e+01 7.648174e+01 7.674301e+01
HQ(n)  7.628643e+01 7.668687e+01 7.708809e+01 7.760199e+01 7.812999e+01
SC(n)  7.675767e+01 7.755082e+01 7.834474e+01 7.925135e+01 8.017205e+01
FPE(n) 9.819799e+32 1.126946e+33 1.302127e+33 1.699217e+33 2.276689e+33
                  6            7            8            9           10
AIC(n) 7.699193e+01 7.723591e+01 7.737594e+01 7.747201e+01 7.756892e+01
HQ(n)  7.864563e+01 7.915634e+01 7.956309e+01 7.992589e+01 8.028953e+01
SC(n)  8.108040e+01 8.198381e+01 8.278327e+01 8.353877e+01 8.429511e+01
FPE(n) 3.061651e+33 4.181450e+33 5.280749e+33 6.590796e+33 8.572216e+33



### <font color=purple> **intérpretation:**

Les critères de sélection des retards (AIC, HQ, SC, et FPE) ont tous indiqué que <font color=red> le nombre optimal de retards pour le modèle est 1,<font color=black> ce qui signifie que les variables explicatives doivent inclure les valeurs retardées d’une période<font color=red> (lag 1).

 ### <font color=purple> **Construction du modèle VAR** 
Une fois les séries stationnaires et le lag optimal déterminé, on peut construire le modèle VAR. Ce modèle explique chaque variable en fonction des valeurs passées de toutes les variables, y compris elle-même.

In [20]:
# Ajustement du modèle VAR avec le lag optimal (1)
var_model <- VAR(data1, p = 1, type = "const")


In [22]:
# Résumé du modèle
summary(var_model)


VAR Estimation Results:
Endogenous variables: Continente, Dis.Mag.Value, nombre.total.de.décès, Total.Affecté, Total.des.dommages...000.US... 
Deterministic variables: const 
Sample size: 107 
Log Likelihood: -4775.82 
Roots of the characteristic polynomial:
0.2005 0.2005 0.111 0.111 0.1021
Call:
VAR(y = data1, p = 1, type = "const")


Estimation results for equation Continente: 
Continente = Continente.l1 + Dis.Mag.Value.l1 + nombre.total.de.décès.l1 + Total.Affecté.l1 + Total.des.dommages...000.US....l1 + const 

                                    Estimate Std. Error t value Pr(>|t|)
Continente.l1                      4.392e-02  1.041e-01   0.422    0.674
Dis.Mag.Value.l1                   1.401e-01  1.123e-01   1.247    0.215
nombre.total.de.décès.l1           4.594e-07  4.594e-06   0.100    0.921
Total.Affecté.l1                  -1.807e-08  1.736e-07  -0.104    0.917
Total.des.dommages...000.US....l1  8.380e-08  5.142e-08   1.630    0.106
const                              1.013

### <font color=purple> Intérpretation:
 Coéfficients des variables explicatives dans l'équation de Dis_Mag_Value

Dans l'équation de Dis.Mag.Value, les variables explicatives incluent Continente.l1, Dis.Mag.Value.l1, nombre.total.de.décès.l1, Total.Affecté.l1, et Total.des.dommages...000.US....l1.

Les résultats des coefficients et des tests de significativité sont les suivants :

Continente.l1 (lag 1) : Le coefficient est estimé à 0.1579, et le test t indique une valeur p de 0.117, <font color=red>ce qui suggère que cette variable n'est pas significativement liée à Dis.Mag.Value (au niveau de signification de 5%). Cela signifie que la région continentale n'a pas d'effet significatif sur la valeur des catastrophes dans ce modèle.

Dis.Mag.Value.l1 (lag 1) : Le coefficient est de 0.0599, avec une valeur p de 0.579,<font color=red> ce qui indique également qu'il n'y a pas d'effet significatif de la valeur de la catastrophe dans la période précédente sur la valeur actuelle de la catastrophe.

nombre.total.de.décès.l1 (lag 1) : Le coefficient est de 0.000005732, avec une valeur p de 0.196, <font color=red>ce qui indique que cette variable n'a pas un effet statistiquement significatif sur Dis.Mag.Value à un niveau de 5%.

Total.Affecté.l1 (lag 1) : Le coefficient est de -0.00000002595, avec une valeur p de 0.876,<font color=red> indiquant que le nombre de personnes affectées dans la période précédente n'a pas d'impact significatif sur la valeur de la catastrophe actuelle.

Total.des.dommages...000.US....l1 (lag 1) : Le coefficient est de -0.00000001845, avec une valeur p de 0.709,<font color=red> ce qui montre également qu'il n'y a pas de relation significative entre le total des dommages en dollars dans la période précédente et la valeur de la catastrophe actuelle.


**=>Aucune des variables explicatives nombre total de décès, total affecté, et total des dommages n'a une relation significative avec Dis.Mag.Value selon les résultats de ce modèle VAR. Toutes ces variables ont des valeurs p supérieures à 0.05, ce qui signifie qu'elles n'ont pas d'impact statistiquement significatif sur la valeur de la catastrophe dans le modèle considéré.
Cela peut indiquer que, dans le contexte des données utilisées, la valeur des catastrophes n'est pas directement influencée par ces facteurs dans la période suivante. Il peut être nécessaire de réexaminer le modèle en ajoutant d'autres variables ou en modifiant les lags (retards) pour mieux capturer les relations entre ces variables.**

# **<font color=red>Conclusion Globale**

Dans le cadre de ce projet de prévision, il est essentiel de souligner que la performance d'un modèle de prévision dépend directement de la pertinence et de l'adaptation des modèles aux données disponibles, ainsi qu'à la nature de la variable cible. Chaque ensemble de données possède des caractéristiques uniques (par exemple, distribution des variables, présence de valeurs manquantes ou extrêmes, etc.) et des objectifs de prévision spécifiques. En conséquence, il est crucial de choisir un modèle qui soit bien adapté à ces spécificités, qu'il s'agisse d'une régression, d'un modèle de classification ou d'un autre type d'algorithme.

Lors de l'évaluation des modèles, il est primordial d'examiner les résultats des tests d'hypothèses ainsi que les métriques de performance (telles que la précision, le rappel, l'AUC, etc.). Si ces tests révèlent que les hypothèses sous-jacentes du modèle sont invalides ou que les métriques sont insatisfaisantes, il est nécessaire d'envisager des corrections ou des ajustements. Ces ajustements peuvent prendre la forme de modifications dans le prétraitement des données, l'ajout ou la suppression de variables explicatives, l'amélioration de la qualité des données ou même la réévaluation des choix méthodologiques.

Dans certains cas, lorsque les résultats restent insuffisants malgré des ajustements, il peut être nécessaire de changer complètement de modèle. Par exemple, un modèle qui ne capte pas correctement la structure des données pourrait être remplacé par un autre plus performant, comme un modèle de machine learning plus complexe (forêts aléatoires, gradient boosting, réseaux de neurones, etc.) ou un modèle adapté à la nature de la variable cible (si elle est catégorielle ou continue).

En conclusion, le choix du modèle doit être un processus dynamique et itératif. Il doit être ajusté en fonction des données disponibles, des hypothèses de travail et des résultats des tests. L'objectif ultime étant d'obtenir un modèle robuste, fiable et capable de fournir des prévisions précises et significatives.


# **<center><font color=red>FIN**
### **<center><font color=purple>Merci pour votre attention!**